PostHog/PostHogContext.swift

```diff
@@ -161,4 +153,94 @@ class PostHogContext {
 
         return properties
     }
+
+    private func registerNotifications() {
+        #if os(iOS) || os(tvOS)
+            #if os(iOS)
+                NotificationCenter.default.addObserver(self,
+                                                       selector: #selector(cacheScreenSize),
+                                                       name: UIDevice.orientationDidChangeNotification,
+                                                       object: nil)
+            #endif
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(cacheScreenSize),
+                                                   name: UIWindow.didBecomeKeyNotification,
+                                                   object: nil)
+        #elseif os(macOS)
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(cacheScreenSize),
+                                                   name: NSWindow.didBecomeKeyNotification,
+                                                   object: nil)
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(cacheScreenSize),
+                                                   name: NSWindow.didChangeScreenNotification,
+                                                   object: nil)
+        #elseif os(watchOS)
+            if #available(watchOS 7.0, *) {
+                NotificationCenter.default.addObserver(self,
+                                                       selector: #selector(cacheScreenSize),
+                                                       name: WKApplication.didBecomeActiveNotification,
```

Comment:
I don't think we have those in the `PostHogSDK#registerNotifications`
So likely missing those events in there as well, mind copying the missing ones?